index.js

```diff
@@ -78,6 +78,8 @@ function retryRequest(requestOpts, opts, callback) {
       } else if (requestStream.end) {
         requestStream.end();
       }
+      
+      requestStream.removeAllListeners();
```

Comment:
I realized converting `on` to `once` alone wouldn't be a fix. I think what we really want is to only run one handler per requestStream. That not being possible with native EventEmitters, as far as I know, the next best thing is just un-registering those events with this line. Let me know if this sounds sane or not.